### PR TITLE
test(option-selectors): implement unit-tests for selectors

### PR DESCRIPTION
**Pull request by Helias**

### CODE REVIEW

File: libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  standalone: true,
+  imports: [GenericOptionSelectorComponent],
+})
+class TestHostComponent {
+  @ViewChild(GenericOptionSelectorComponent) child!: GenericOptionSelectorComponent;
+  mockFormControl = new FormControl();
+  EXPANSION = EXPANSION;
+}
+
+class GenericOptionSelectorComponentPage extends PageObject<TestHostComponent> {}
```
Comment (Francesco-Borzi) [inline libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts:19]: same

File: libs/shared/selectors/src/selectors/boolean-option-selector/boolean-option-selector.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const host = fixture.componentInstance;
+    const component = host.child;
+    const page = new BooleanOptionSelectorComponentPage(fixture);
+
+    fixture.detectChanges();
+
+    return { fixture, host, component, page };
+  };
+
+  it('change select values', () => {
```
Comment (Francesco-Borzi) [inline libs/shared/selectors/src/selectors/boolean-option-selector/boolean-option-selector.component.spec.ts:37]:
```suggestion
  it('changing the form value will be reflected in the template', () => {
```

File: libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const host = fixture.componentInstance;
+    const component = host.child;
+    const page = new GenericOptionSelectorComponentPage(fixture);
+
+    fixture.detectChanges();
+
+    return { fixture, host, component, page };
+  };
+
+  it('change select values', () => {
```
Comment (Francesco-Borzi) [inline libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts:39]:
```suggestion
  it('changing the form value will be reflected in the template', () => {
```

File: libs/shared/selectors/src/selectors/boolean-option-selector/boolean-option-selector.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    expect(select.nativeElement.value).toEqual('1: 1');
+    expect(select.nativeElement.selectedOptions[0].innerText).toEqual('1 - SELECTORS.ENABLED');
+
+    host.mockFormControl.setValue(0);
+    page.detectChanges();
+
+    expect(select.nativeElement.value).toEqual('0: 0');
+    expect(select.nativeElement.selectedOptions[0].innerText).toEqual('0 - SELECTORS.DISABLED');
+  });
+});
```
Comment (Francesco-Borzi) [inline libs/shared/selectors/src/selectors/boolean-option-selector/boolean-option-selector.component.spec.ts:53]: add the opposite test: changing the form value via template will be reflected in the input property

File: libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    console.log('select.nativeElement.selectedOptions', select.nativeElement.selectedOptions);
+    expect(select.nativeElement.selectedOptions[0].label).toEqual('1 - The Burning Crusade');
+
+    host.mockFormControl.setValue(0);
+    page.detectChanges();
+
+    expect(select.nativeElement.value).toEqual('0: 0');
+    expect(select.nativeElement.selectedOptions[0].label).toEqual('0 - Classic');
+  });
+});
```
Comment (Francesco-Borzi) [inline libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts:56]: add the opposite test: changing the form value via template will be reflected in the input property

**Review by Francesco-Borzi — APPROVED** [review]

**Review by FrancescoBorzi — APPROVED** [review]

File: libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    expect(select.value).toEqual('0: 0');
+    expect(select.selectedOptions[0].label).toEqual('0 - Classic');
+  });
+
+  it('changing the input valuewill be reflected in the form control', () => {
+    const { page, host } = setup();
+    const select = page.getDebugElementByCss('select').nativeElement;
+
+    select.value = '1: 1';
+    select.dispatchEvent(new Event('change'));
```
Comment (FrancescoBorzi) [inline libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts:61]: there is already a helper method in the PageObject class that does this

Comment (Helias) [reply libs/shared/selectors/src/selectors/generic-option-selector/generic-option-selector.component.spec.ts:61]: I will handle this in another PR

**Pull request merged**